config/connectionAPI/secret_s3_validation.yaml

```diff
@@ -0,0 +1,37 @@
+---
+# ValidatingAdmissionPolicy to warn when S3 connection secrets are missing AWS_S3_BUCKET
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: connectionapi-check-s3-bucket
+spec:
+  failurePolicy: Fail
+  matchConstraints:
+    resourceRules:
+      - apiGroups: [""]
+        apiVersions: ["v1"]
+        operations: ["CREATE", "UPDATE"]
+        resources: ["secrets"]
```

Comment:
@zdtsw It looks like this VAP is going to run when any secret across the entire cluster is created or updated which may impact performance. Can we use `.spec.matchConstraints.objectSelector.matchLabels` to narrow the scope?

Reply:
performance wise i guess it is ok, it wont check any existing ones, and only on create, update. similar to our webhook on certain type

Reply:
Won't the s3 connection secret have either the deprecated `opendatahub.io/connection-type-ref` annotation or the new `opendatahub.io/connection-type-protocol` annotation?

Reply:
removed